### PR TITLE
ghost bar members are now cleared more hastily

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
ghost bar members are now cleared more hastily. You'll now have your previous mob cleared if you rejoin the bar

## Why It's Good For The Game
bugfix

## Testing
did it in game

## Changelog
:cl:
fix: Ghost bar members are now cleared when rejoining the bar with a character already in the bar
/:cl:

### CODE REVIEW

**Comment by Scribble-Sheep** [comment]
While I'm sad that it's harder to show people all my characters side by side, I get why this was done. Holy shit lol

File: code/modules/ruins/ghost_bar.dm [hunk 1]
```diff
@@ -83,13 +84,21 @@
...
+	RegisterSignal(H, COMSIG_PARENT_QDELETING, PROC_REF(clear_references_to_owner))
 
 /obj/effect/mob_spawn/human/alive/ghost_bar/proc/equip_item(mob/living/carbon/human/H, path, slot)
 	var/obj/item/I = new path(H)
 	H.equip_or_collect(I, slot, TRUE)
 	H.speaks_ooc = TRUE
 	return I
 
+/obj/effect/mob_spawn/human/alive/ghost_bar/proc/clear_references_to_owner(mob/mob_to_obliterate)
+	GLOB.occupants_by_key -= mob_to_obliterate.ckey
```
Comment (lewcc) [inline code/modules/ruins/ghost_bar.dm:100]:
```suggestion
	SIGNAL_HANDLER  // COMSIG_PARENT_QDELETING
	GLOB.occupants_by_key -= mob_to_obliterate.ckey
```

File: code/modules/ruins/ghost_bar.dm [hunk 2]
```diff
@@ -55,7 +56,7 @@
 		equip_item(H, /obj/item/clothing/under/color/random, SLOT_HUD_JUMPSUIT)
 	if(!H.shoes)
 		equip_item(H, /obj/item/clothing/shoes/black, SLOT_HUD_SHOES)
-	equip_item(H, /obj/item/stack/spacecash/c1000, SLOT_HUD_LEFT_STORE)
+	equip_item(H, /obj/item/stack/spacecash/c100, SLOT_HUD_LEFT_STORE)
```
Comment (lewcc) [inline code/modules/ruins/ghost_bar.dm:59]: Unrelated change?

Comment (GDNgit) [reply code/modules/ruins/ghost_bar.dm:59]: yeah just conflict funnies

Comment (GDNgit) [inline code/modules/ruins/ghost_bar.dm:59]:
```suggestion
	equip_item(H, /obj/item/stack/spacecash/c1000, SLOT_HUD_LEFT_STORE)
```

Comment (Scribble-Sheep) [reply code/modules/ruins/ghost_bar.dm]: Accidental stealth revert, huh? Lmao

File: code/modules/ruins/ghost_bar.dm [hunk 3]
```diff
@@ -1,3 +1,4 @@
+GLOBAL_LIST_EMPTY(occupants_by_key)
 /obj/effect/mob_spawn/human/alive/ghost_bar
```
Comment (DGamerL) [inline code/modules/ruins/ghost_bar.dm:2]:
```suggestion
GLOBAL_LIST_EMPTY(occupants_by_key)

/obj/effect/mob_spawn/human/alive/ghost_bar
```

**Review by DGamerL — APPROVED** [review]

**Review by Burzah — APPROVED** [review]
